
### Examining racial discrimination in the US job market

#### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

#### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes.

#### Exercise
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Discuss statistical significance.

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [41]:
import pandas as pd
import numpy as np
from scipy import stats

In [42]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [43]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [44]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...            1       0      1         0         0     0   
1            316    ...            1       0      1         0         0     0   
2             19    ...            1       0      1         0         0     0   
3            313    ...            1       0      1         0         0     0   
4            313    ...            1       1      0         0         0     0   

  busservice othservice  missind  ownership  
0          0          0        0             
1          0          0        0             
2          0          0        0             
3          0          0        0             
4          0          1        0  Nonprofit  

[5 rows x 65 columns]

In [45]:
data.id.count()

4870

This data set has 4870 resumes.

In [46]:
bcandidates = data[data.race=='b']
bcount = bcandidates.id.count()
bcount

2435

Exactly half of the resumes have black-sounding names.

In [47]:
bcalled = sum(data[data.race=='b'].call)
bcalled

157.0

In [48]:
wcalled = sum(data[data.race=='w'].call)
wcalled

235.0

In [49]:
bcalled+wcalled

392.0

A total of 392 applicants (235 white and 157 black) received callbacks out of the 4870 who applied.

The populations in question here are the populations of black and white people. The property that may be assumed to be normally distributed is the suitability for the job that applications were sent for.

Therefore, our sample size is 2435 from each of the populations. 

The Central Limit Theorem can be used here because:
1. The data is sampled randomly from the populations
2. The sample values are independent of each other
3. The number of samples is sufficiently large

The null hypothesis is that the percentage of candidates called back is independent of the race of the candidate.

The alternate hypothesis is that being black reduces the likelihood of candidates receiving callbacks.

The mean of these distributions can be understood as the % of candidates who received callbacks.

In [50]:
bmean = bcalled/bcandidates.id.count()
bmean

0.064476386036960986

In [51]:
wmean = wcalled/bcandidates.id.count()
wmean

0.096509240246406572

Since these are Bernoulli distributions - the variance is $$p(1-p)/n$$

where $$p$$ is the mean of the samples, since we may assume that an outcome of 1 corresponds to a callback, and 0 to no callback. The standard deviations are then:

In [52]:
bstd = np.sqrt(bmean*(1-bmean)/bcount)
wstd = np.sqrt(wmean*(1-wmean)/bcount)
[bstd, wstd]

[0.0049771214428119461, 0.0059840721781280661]

Let's now consider the difference of the two distributions. The mean of the difference distribution is

In [53]:
diff_mean = wmean - bmean
diff_mean

0.032032854209445585

The SD of this distribution is

In [54]:
diff_sd = np.sqrt(np.square(bstd)/bcount+np.square(wstd)/bcount)
diff_sd

0.00015773142469580601

As in assignment 1, the 95% confidence interval corresponds to 1.96 times the SD

In [55]:
[diff_mean-1.96*diff_sd, diff_mean+1.96*diff_sd]

[0.031723700617041807, 0.032342007801849364]

We may calculate the range for a 99% confidence interval as:

In [56]:
[diff_mean-2.5758*diff_sd, diff_mean+2.5758*diff_sd]

[0.031626569605714125, 0.032439138813177046]

This indicates that there is less than a 1% chance that the differences in callbacks between black and white applicants can be explained by randomness. In other words, this is evidence that there is a bias against black applicants.